pkg/cpio/cpio.go

```diff
@@ -77,9 +78,28 @@ func (r Reader) ReadRecords() ([]Record, error) {
 
 type Writer struct {
 	rw RecordWriter
+	// There seems to be no harm done in stripping
+	// duplicate names when the record is written,
+	// and lots of harm done if we don't do it.
+	alreadyWritten map[string] bool
 }
 
 func (w Writer) WriteRecord(rec Record) error {
+	// we do NOT write records with absolute paths.
+	if filepath.IsAbs(rec.Name) {
```

Comment:
all of our records are generated with cpio.GetRecord or created manually. This would suggest that cpio.GetRecord is wrong, not cpio.Write?